backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java

```diff
@@ -618,4 +625,27 @@ private void updateVersionNumber(Handle handle, String expId) {
             executor)
         .thenCompose(unused -> getExperimentById(expId), executor);
   }
+
+  public InternalFuture<GetAttributes.Response> getExperimentAttributes(GetAttributes request) {
+    final var expId = request.getId();
+    final var keys = request.getAttributeKeysList();
+    final var getAll = request.getGetAll();
+
+    return getProjectIdByExperimentId(Collections.singletonList(expId))
+        .thenCompose(
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(expId), ModelDBServiceActions.READ),
+            executor)
+        .thenCompose(unused -> attributeHandler.getKeyValues(expId, keys, getAll), executor)
+        .thenApply(
+            attributes ->
+                attributes.stream()
+                    .sorted(Comparator.comparing(KeyValue::getKey))
+                    .collect(Collectors.toList()),
+            executor)
```

Comment:
`attributeHandler` should handle this, no?

Reply:
Yes it should. 
Jira ticket: https://vertaai.atlassian.net/browse/VUMM-207